swatch-system-conduit/src/main/java/org/candlepin/subscriptions/conduit/InventoryController.java

```diff
@@ -330,6 +333,70 @@ protected void extractIpAddresses(Map<String, String> rhsmFacts, ConduitFacts fa
     }
   }
 
+  private List<NetworkInterface> populateNICs(Map<String, String> rhsmFacts) {
+    var nicSet = new ArrayList<NetworkInterface>();
+    for (Map.Entry<String, String> entry : rhsmFacts.entrySet()) {
+      if (entry.getKey().startsWith(MAC_PREFIX) && entry.getKey().endsWith(MAC_SUFFIX)) {
+        String[] nicsName = entry.getKey().split(PERIOD_REGEX);
+        var mac = entry.getValue();
+        var networkInterface = new NetworkInterface();
+        networkInterface.setName(nicsName[2]);
+        networkInterface.setMacAddress(mac);
+        mapInterfaceIps(networkInterface, rhsmFacts, ".ipv4");
+        mapInterfaceIps(networkInterface, rhsmFacts, ".ipv6");
+        nicSet.add(networkInterface);
+      }
+    }
+    // creates a lo interface if ips exist for it, but no mac was given
+    checkLoopbackIPs(nicSet, rhsmFacts);
+    return nicSet;
+  }
+
+  private void mapInterfaceIps(
+      NetworkInterface networkInterface, Map<String, String> facts, String suffix) {
+    String regex = MAC_PREFIX + networkInterface.getName() + suffix;
```

Comment:
Nitpick: this isn't a regex; I'd prefer a different variable name, such as `prefix`.